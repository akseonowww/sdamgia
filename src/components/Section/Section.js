import React from 'react';

import './Section.css';

const Section = props => {
	const { className, children } = props;

	return <section className={`Section ${className}`}>{children}</section>;
};

export default Section;
