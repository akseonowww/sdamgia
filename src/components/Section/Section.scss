@import 'src/styles/variables', 'src/styles/breakpoints';

.Section {
	margin-bottom: 1.75rem;

	font-size: 1rem;

	border-top: 0.15rem solid $primary;

	&-ComplexHeader {
		display: flex;
		flex-direction: column;

		@include media-breakpoint-up(md) {
			flex-direction: row;
			justify-content: space-between;
		}
	}

	&-Desc {
		margin: 0 0 1rem;

		line-height: 1.4rem;
		text-align: left;
		text-indent: 1.5rem;

		@include media-breakpoint-up(md) {
			padding-right: 1.75rem;
		}
	}
}
