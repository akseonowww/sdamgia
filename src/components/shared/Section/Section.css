.Section {
	margin-bottom: 25px;

	font-size: 10pt;

	border-top: 2px solid #006;
}

.Section-ComplexHeader {
	display: flex;
	justify-content: space-between;
	margin-top: 0;
}

.Section-Desc {
	margin: 0 0 10px;
	padding-right: 25px;

	line-height: 13pt;
	text-align: left;
}
