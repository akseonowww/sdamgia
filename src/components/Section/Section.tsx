import React, { FC, ReactNode } from 'react'
import cx from 'classnames'

import './Section.scss'

interface ISectionProps {
  className?: string
  children: ReactNode
}

const Section: FC<ISectionProps> = ({ className, children }) => (
  <section className={cx('Section', className)}>{children}</section>
)

export default Section
