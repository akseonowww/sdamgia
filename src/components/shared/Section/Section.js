import React from 'react';

import './Section.css';

const Section = props => (
	<section className="Section">{props.children}</section>
);

export default Section;
