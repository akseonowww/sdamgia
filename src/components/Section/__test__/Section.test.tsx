import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import Section from '../Section'

describe('Section', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with or without a className', () => {
    act(() => {
      render(
        <Section>
          <div>Auxiliary Section</div>
        </Section>,
        container
      )
    })
    expect(container && container.textContent).toBe('Auxiliary Section')

    act(() => {
      render(
        <Section className="Student-Section">
          <div>Student Section</div>
        </Section>,
        container
      )
    })
    expect(container && container.textContent).toBe('Student Section')
  })

  it('should render a content', () => {
    act(() => {
      render(
        <Section>
          <div>Auxiliary Section</div>
        </Section>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <Section className="Student-Section">
          <div>Student Section</div>
        </Section>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <Section className="Student-Section">
          <h1>Student Section</h1>
          <div>Description</div>
        </Section>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
