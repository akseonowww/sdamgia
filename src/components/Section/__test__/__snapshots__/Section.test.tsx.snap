// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render a content 1`] = `
"<section class=\\"Section\\">
  <div>Auxiliary Section</div>
</section>"
`;

exports[`Section should render a content 2`] = `
"<section class=\\"Section Student-Section\\">
  <div>Student Section</div>
</section>"
`;

exports[`Section should render a content 3`] = `
"<section class=\\"Section Student-Section\\">
  <h1>Student Section</h1>
  <div>Description</div>
</section>"
`;
